pkg/storage/storage.go

```diff
@@ -133,17 +134,22 @@ func TryLoadRegionsOnce(ctx context.Context, s Storage, f func(region *core.Regi
 		return s.LoadRegions(ctx, f)
 	}
 
+	ps.mu.Lock()
+	defer ps.mu.Unlock()
+
 	if atomic.LoadInt32(&ps.useRegionStorage) == 0 {
-		return ps.Storage.LoadRegions(ctx, f)
+		err := ps.Storage.LoadRegions(ctx, f)
+		if err == nil {
+			ps.regionLoadedFromDefault = true
+		}
+		return err
 	}
 
-	ps.mu.Lock()
-	defer ps.mu.Unlock()
-	if !ps.regionLoaded {
+	if !ps.regionLoadedFromStorage {
 		if err := ps.regionStorage.LoadRegions(ctx, f); err != nil {
 			return err
 		}
-		ps.regionLoaded = true
+		ps.regionLoadedFromStorage = true
 	}
 	return nil
```

Comment:
Seems we can use `*coreStorage.LoadRegions()` directly?

Reply:
It doesn't check ps.regionLoadedFromStorage